src/PrestaShopBundle/Controller/ArgumentResolver/FiltersBuilderResolver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\ArgumentResolver;
+
+use PrestaShop\PrestaShop\Core\Search\Builder\FiltersBuilderInterface;
+use PrestaShop\PrestaShop\Core\Search\Filters;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * This argument resolver uses the FiltersBuilderInterface service to automatically
+ * instantiate and inject parameters in controllers.
+ */
+class FiltersBuilderResolver implements ArgumentValueResolverInterface
+{
+    /** @var FiltersBuilderInterface */
+    private $builder;
+
+    /**
+     * @param FiltersBuilderInterface $builder
+     */
+    public function __construct(FiltersBuilderInterface $builder)
+    {
+        $this->builder = $builder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument)
+    {
+        return is_subclass_of($argument->getType(), Filters::class);
```

Comment:
Oooooh 😮 I did not know about `is_subclass_of` !

Reply:
Me neither 😅